presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewInformationExtractor.java

```diff
@@ -77,35 +78,34 @@ protected Void visitRelation(Relation node, Void context)
     @Override
     protected Void visitSingleColumn(SingleColumn node, Void context)
     {
+        process(node.getExpression());
         Expression baseColumnName = node.getExpression();
         materializedViewInfo.addBaseToViewColumn(baseColumnName, node.getAlias().orElse(new Identifier(baseColumnName.toString())));
         return null;
     }
 
     @Override
-    protected Void visitAllColumns(AllColumns node, Void context)
-    {
-        throw new SemanticException(NOT_SUPPORTED, node, "All columns materialized view is not supported in query optimizer");
-    }
-
-    @Override
-    protected Void visitGroupBy(GroupBy node, Void context)
+    protected Void visitFunctionCall(FunctionCall node, Void context)
     {
-        materializedViewInfo.setGroupBy(Optional.of(ImmutableSet.copyOf(node.getGroupingElements())));
+        if (node.getName().equals(QualifiedName.of("COUNT"))) {
```

Comment:
Can we have list of unsupported function as a configuration, so that we don't have to hardcode?